sleap/gui/web.py

```diff
@@ -146,6 +149,49 @@
         )
 
 
+@attr.s(auto_attribs=True)
+class AnnouncementChecker:
+    """Checker for new announcements on the bulletin page of sleap."""
+
+    bulletin_json_path: str = BULLETIN_JSON
+    previous_announcement_date: str = prefs.prefs["announcement last seen date"]
+
+    def check_for_announcement(self) -> bool:
+        """Returns if new announcements are available."""
+        try:
+            # Attempt to open the file in read mode
+            with open(self.bulletin_json_path, 'r', encoding='utf-8') as jsf:
+                # Load the JSON data into a Python data structure
+                data = json.load(jsf)
+                latest_data = data[0]
+                
+                if latest_data['date'] != self.previous_announcement_date:
+                    return True
+        except FileNotFoundError:
+            return False
```

Comment:
@coderabbitai This will eventually be a ping yielding an online JSON response, would that change your response? Otherwise, what would you suggest with this new information?